### ISSUE TITLE
Drag & Drop mechanism is inconsistent between Chrome 95 and 96 and other browsers. 

### ISSUE THREAD

**Issue by pmario** [codemirror/CodeMirror#6831]
To reproduce: 

 1. With **FireFox** 94.0.2
 2. open https://codemirror.net/
 3. Clear the "This is CodeMirror" demo text input
 4. Drag the eg: "theming" link above the text input into the text input
 5. you get: `https://codemirror.net/doc/manual.html#styling` **as expected**

--- 

 - Now do the same thing with **Chrome** or **Edge** 96 latest

In step 5) you'll get the wrong thing:

```
[InternetShortcut]
URL=https://codemirror.net/doc/manual.html#styling
```

expected:

```
https://codemirror.net/doc/manual.html#styling
```

-----

 - Now drag and drop the same link from the **FireFox** window
 - to the **Edge** "text editor"

You'll get: 

```
[InternetShortcut]
URL=https://codemirror.net/doc/manual.html#styling
IDList=
HotKey=0
IconFile=...\Users\....\shortcutCache\...==.ico
IconIndex=0
```

exptected: `https://codemirror.net/doc/manual.html#styling`

-----------

From one of our contributors we got this info: 

>A very quick look shows that the drop event received in Chrome 96 now has an entry in dataTransfer.files which has previously not been the case, and in our handling of the drop events this takes precedence over dataTransfer.items.
>
>![grafik](https://user-images.githubusercontent.com/374655/143251403-e0cb4073-8fa8-4215-b312-5580c7a208e5.png)
>
>Edit: a workaround would be easy to add but I am hesitant to do so without understanding the reasoning behind the change in Chrome. A quick search has not turned up any relevant Chrome documentation or change history.

We have an easy fix for the browser default text-area, since we can control the "drop serializer", but in combination with CM it's a different topic.

**Comment by Jermolene** [comment]
Hi @pmario can this be closed?

**Comment by pmario** [comment]
>Hi @pmario can this be closed?

@Jermolene ... imo not really, the problem still exists as described in the OP.

**Comment by marijnh** [comment]
Chrome 66 was released in 2018 and is probably not used by any significant amount of people anymore. Chrome 92 doesn't seem to have this issue. Given that, why are you reporting this at this time?

**Comment by Jermolene** [comment]
Apologies @marijnh it's a typo in the OP – this is a problem with the latest Chrome **96**, not Chrome 66

**Comment by pmario** [comment]
Sorry. Chrome  96

**Comment by marijnh** [comment]
Not seeing this with Chromium 96.0.4664.0 on Linux.

**Comment by Jermolene** [comment]
> Not seeing this with Chromium 96.0.4664.0 on Linux.

I think it's Windows only. I wasn't able to duplicate on macOS either.

**Comment by marijnh** [comment]
Chrome 96.0.4664.110 in my Windows 7 VM also doesn't show the issue either.

**Comment by gjveld** [comment]
On Linux - Chrome 96.0.4664.93 (Official Build) (64-bit) drag and drop stopped working at all

**Comment by marijnh** [comment]
Works fine for me in  96.0.4664.93

**Comment by Jermolene** [comment]
I've now had a chance to set up a VM with Chrome 96.0.4664.93 on Windows 10, and was also unable to duplicate the problem.

@pmario do you have the exact version of Chrome where you saw the problem?

**Comment by pmario** [comment]
@Jermolene As in the OP I did test it with Edge on my **Windows 10** production PC

TLDR;

I did test the whole thing with the WindowsSandbox and installed Chrome Version 96.0.4664.110 (Official Build) (64-bit), FireFox 95.0.1 and Edge Version 96.0.1054.57 (Official build) (64-bit) .... No plugins installed.

I can't reproduce the problem there, since I can't d&d a link from FireFox to Chrome or Edge ... 

So it may have something to do with my production system settings. 

I'll close the issue, since the problem seems to be fixed with our latest TiddlyWiki version.

------

Tests with my production Windows 10

 - Open https://codemirror.net/ with FireFox 95.0.1 
 - Open codemirror.net with Edge -- Version 96.0.1054.57 (Official build) (64-bit) 
 - drag the "theming" link from FireFox to Edge editor "This is CodeMirror" 

I did get:

```
[InternetShortcut]
URL=https://codemirror.net/doc/manual.html#styling
IDList=
HotKey=0
IconFile=C:\Users\.......\Mozilla\Firefox\Profiles\xxxxxxx.default-release\shortcutCache\1.....==.ico
IconIndex=0
```

**expected**

```
https://codemirror.net/doc/manual.html#styling
```

-----

 - Open codemirror.net with Edge
 - Drag the "theming" link in Edge to the editor
 - I did get the expected result.

The same is true, if I drag & drop the link with FireFox to the editor in Firefox.

**Comment by gjveld** [comment]
I thought it to be a bug with CodeMirror. But it is, in my view, a bug with chrome. Which also effects CodeMirror.

I made a codepen to demonstrate it.
It's not the standard drag and drops (like DND a link) but the ones you manipulate the data through the Javascript DND APi.

**Comment by pmario** [comment]
>Do I see/ do it wrong? Or is it really a chrome bug?
It works in firefox...

You may be right.
